cpu/stm32l1/periph/i2c.c

```diff
@@ -466,100 +362,105 @@ int i2c_write_regs(i2c_t dev, uint8_t address, uint8_t reg, char *data, int leng
 
 void i2c_poweron(i2c_t dev)
 {
-    switch (dev) {
-#if I2C_0_EN
-        case I2C_0:
-            I2C_0_CLKEN();
-            break;
-#endif
-#if I2C_1_EN
-        case I2C_1:
-            I2C_1_CLKEN();
-            break;
-#endif
-    }
+    RCC->APB1ENR |= (RCC_APB1ENR_I2C1EN << dev);
```

Comment:
That was what I (generally) meant with removing the CLK enabling macros in the periph_conf - yes. But are you sure that this piece of code does what you expect? I'm not but I didn't check in detail